Task5 - Java Servlets

Paveline opened the pull request:
Good day! This is the new purpose of java servlets!

P.S.
When creating servlets, I took advantage of the new functionality and created them using @WebServlet instead of using the xml file, because it is much more convenient, less code and looks prettier!

Paveline committed:
Add Task5 - Java Servlets

smiakchilo commented on task5/java_servlets/demo_servlets.iml:
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
This one's IDEA's tech file, it should be `gitignore`d

smiakchilo commented on task5/java_servlets/demo_servlets.iml:
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
```
All the `.class` files must be `gitignore`d as well)

smiakchilo commented on task5/java_servlets/src/java/com/CheckServlet.java:
```diff
@@ -0,0 +1,29 @@
+            String response = "{\"sucess\":true}";
+            writer.println(response);
+        } catch (Exception e) {
+            System.out.println("error");
```
In real life, you'd better print out the `e` as is (it would be converted into a meaningful string) or the message by `e.getMessage()`. 
By printing the "error" word you are just giving away the very possibility to get know what in particular has happened

smiakchilo commented on task5/java_servlets/src/java/com/CheckServlet.java:
```diff
@@ -0,0 +1,29 @@
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        doGet(req, resp);
```
This is not the right way, it spoils the REST principle. Post requests are never equal to Get requests.
You can easily skip this method altogether. It has an implementation in a superclass, therefore you don't need to override it in a subclass

smiakchilo commented on task5/java_servlets/src/java/com/GetNameServlet.java:
```diff
@@ -0,0 +1,21 @@
+        if (name == null || name.length() == 0)
+            name = "No name";
+
+        resp.getOutputStream().println("<html><h2>" + name + "</h2></html>");
```
What about the name greater than 100 chars?

smiakchilo reviewed (changes requested):
Please remove the redundant files, then feel free to merge

Paveline commented:
Good day! Thanks for the advices!

I only have a couple of questions:
1) Can I delete all these files manually now, and then create a gitignore, or do I need to upload everything to a new branch?

2) Can you please tell me how to correctly indicate the classes in the gitignore, because I did not quite understand

3) It turns out that I need to correct what you wrote in the program and then upload it to this branch again?
 
Thanks again! Have a nice day!)

smiakchilo commented:
1.	Yes you can physically delete them because they are anyway re-created on compilation stage. They are compilation artifacts. They sit in /target folder, and the while /target folder should be gitignored
2.	There is the site. https://gitigonore.io that contains gitignore samples for different languages. You can play around with it… What concerns .class files, it’s enough to specify *.class. But usually people ignore the whole “target” folder – this is the folder created in compilation – and it goes like  **/target/   (two ** means “/target folder, and no matter what folders are above”)
3.	Yes, you do new commits to the same branch, to the same pull-request. Then you can re-request review (there’s the “update” little button next to the reviewers name in the top right corner). Or in this particular case you are allowed to merge after changes)

 

From: Paveline ***@***.***> 
Sent: Tuesday, March 16, 2021 8:18 AM
To: Paveline/UPractica ***@***.***>
Cc: Stepan Miakchilo ***@***.***>; Review requested ***@***.***>
Subject: Re: [Paveline/UPractica] Task5 - Java Servlets (#4)

 

Good day! Thanks for the advices!

I only have a couple of questions:

1.	Can I delete all these files manually now, and then create a gitignore, or do I need to upload everything to a new branch?
2.	Can you please tell me how to correctly indicate the classes in the gitignore, because I did not quite understand
3.	It turns out that I need to correct what you wrote in the program and then upload it to this branch again?

Thanks again! Have a nice day!)

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub <https://github.com/Paveline/UPractica/pull/4#issuecomment-799957165> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ALUI6RBZ5HXP4EHC4SAEBNTTD3SZTANCNFSM4ZG2DRVA> .


-- 


CONFIDENTIALITY
NOTICE: This email and files attached to it are 
confidential. If you
are not the intended recipient you are hereby notified 
that using,
copying, distributing or taking any action in reliance on the 
contents of this information is strictly prohibited. If you have
received 
this email in error please notify the sender and delete this
email.

Paveline committed:
Create .gitignore

Paveline committed:
Delete Out Folder

Paveline committed:
Delete demo_servlets.iml

Paveline committed:
Update Task5

Paveline commented:
Thank you! I have corrected everything you said!

1) An error message is displayed

2) If the string is more than 100 characters long, a substring of 100 characters is taken and a message is displayed that the word length should not exceed 100 characters

3) Moved all the code from the get-method to the post-method (as I understand it, you meant it)

4) Added .gitignore

smiakchilo commented:
Fine!

 

From: Paveline ***@***.***> 
Sent: Tuesday, March 16, 2021 3:06 PM
To: Paveline/UPractica ***@***.***>
Cc: Stepan Miakchilo ***@***.***>; Review requested ***@***.***>
Subject: Re: [Paveline/UPractica] Task5 - Java Servlets (#4)

 

Thank you! I have corrected everything you said!

1.	An error message is displayed
2.	If the string is more than 100 characters long, a substring of 100 characters is taken and a message is displayed that the word length should not exceed 100 characters
3.	Moved all the code from the get-method to the post-method (as I understand it, you meant it)
4.	Added .gitignore

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub <https://github.com/Paveline/UPractica/pull/4#issuecomment-800203781> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ALUI6RBDVPVCRUDNYGR7HSTTD5CRVANCNFSM4ZG2DRVA> .


-- 


CONFIDENTIALITY
NOTICE: This email and files attached to it are 
confidential. If you
are not the intended recipient you are hereby notified 
that using,
copying, distributing or taking any action in reliance on the 
contents of this information is strictly prohibited. If you have
received 
this email in error please notify the sender and delete this
email.
